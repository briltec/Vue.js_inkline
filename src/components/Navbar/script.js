import IContainer from '../Container';
import IRow from '../Row';
import IColumn from '../Column';

import AttributesProviderMixin from '@inkline/inkline/mixins/components/providers/AttributesProviderMixin';
import ClassesProviderMixin from '@inkline/inkline/mixins/components/providers/ClassesProviderMixin';

import SizePropertyMixin from '@inkline/inkline/mixins/components/properties/SizePropertyMixin';
import VariantPropertyMixin from '@inkline/inkline/mixins/components/properties/VariantPropertyMixin';

import { breakpoints } from '@inkline/inkline/constants';

export default {
    name: 'INavbar',
    mixins: [
        AttributesProviderMixin,
        ClassesProviderMixin,

        SizePropertyMixin,
        VariantPropertyMixin
    ],
    components: {
        IContainer,
        IRow,
        IColumn
    },
    provide() {
        const navbar = {};

        Object.defineProperty(navbar, 'collapsed', {
            enumerable: true,
            get: () => this.collapsed,
        });

        Object.defineProperty(navbar, 'collapse', {
            enumerable: true,
            get: () => this.collapse,
        });

        return { navbar };
    },
    data() {
        return {
            collapsed: false,
            windowWidth: window.innerWidth
        };
    },
    props: {
        fluid: {
            type: Boolean,
            default: false
        },
        collapse: {
            type: String,
            default: 'md'
        },
        toggleAnimation: {
            type: String,
            default: 'close'
        }
    },
    methods: {
        toggleCollapse() {
            this.collapsed = !this.collapsed;
        },
        onWindowResize() {
            const windowWidth = window.innerWidth;

            if (this.windowWidth <= breakpoints[this.collapse][1] && windowWidth > breakpoints[this.collapse][1]) {
                this.collapsed = false;
            }

            this.windowWidth = windowWidth;
        }
    },
    created () {
        this.classesProvider.add(() => ({
            '-collapsed': this.collapsed,
            [`-collapse-${this.collapse}`]: Boolean(this.collapse)
        }));

        this.classesProvider.add('hamburger', () => ({
            '-active': this.collapsed,
            [`-${this.toggleAnimation}`]: true
        }));

        window.addEventListener('resize', this.onWindowResize);
        this.onWindowResize();
    },
    destroyed() {
        window.removeEventListener('resize', this.onWindowResize)
    },
};