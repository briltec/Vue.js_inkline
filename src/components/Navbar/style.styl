/**
 * Navbar
 */

@import '../../css/config'
@import '../../css/mixins'
@import 'css/_variables'
@import 'css/_variants'
@import 'css/_sizes'

.navbar
    position: relative
    display: flex
    flex-wrap: wrap // allow us to do the line break for collapsing content
    align-items: center
    justify-content: space-between

    > .container
        display: flex
        flex-wrap: wrap
        align-items: center
        justify-content: space-between

    @extend .navbar.-md
    @extend .navbar.-light