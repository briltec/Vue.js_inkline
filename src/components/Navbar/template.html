<div class="navbar"
     :class="classes['root']"
     v-bind="attributes">
    <i-container :fluid="fluid">
        <i-row>
            <i-column :xs="true">
                <div class="hamburger" :class="classes['hamburger']" @click="toggleCollapse">
                    <span class="bars"></span>
                </div>
                <slot></slot>
            </i-column>
        </i-row>
    </i-container>
</div>