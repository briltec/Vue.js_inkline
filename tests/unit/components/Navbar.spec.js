import { mount } from '@vue/test-utils';
import Navbar from 'inkline/components/Navbar';

describe('Components', () => {
    describe('Navbar', () => {
        let wrapper;

        beforeEach(() => {
            wrapper = mount(Navbar);
        });

        it('should render correctly', () => {
            expect(wrapper.html()).toMatchSnapshot();
        });
    });
});
