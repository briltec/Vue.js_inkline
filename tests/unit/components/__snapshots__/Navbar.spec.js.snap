// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Navbar should render correctly 1`] = `
<div class="navbar">
  <div class="container"></div>
</div>
`;
